src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt

```diff
@@ -0,0 +1,199 @@
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.slf4j.LoggerFactory
+import org.springframework.format.annotation.DateTimeFormat
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.hmpps.prison.api.model.ErrorResponse
+import uk.gov.justice.hmpps.prison.api.model.PrisonerDetail
+import uk.gov.justice.hmpps.prison.api.model.PrisonerDetailSearchCriteria
+import uk.gov.justice.hmpps.prison.api.support.Order
+import uk.gov.justice.hmpps.prison.api.support.PageRequest
+import uk.gov.justice.hmpps.prison.core.SlowReportQuery
+import uk.gov.justice.hmpps.prison.service.GlobalSearchService
+import java.time.LocalDate
+
+@RestController
+@Tag(name = "prisoners")
+@RequestMapping(value = ["\${api.base.path}/prisoners"], produces = ["application/json"])
+@Validated
+class PrisonerResource(private val globalSearchService: GlobalSearchService) {
+  @ApiResponses(
+    ApiResponse(responseCode = "200", description = "OK"),
+    ApiResponse(responseCode = "400", description = "Invalid request.", content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]),
+    ApiResponse(responseCode = "404", description = "Requested resource not found.", content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]),
+    ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]),
+  )
+  @Operation(summary = "List of offenders matching specified criteria.", description = "List of offenders matching specified criteria.")
+  @GetMapping
+  @PreAuthorize("hasAnyRole('SYSTEM_USER','GLOBAL_SEARCH')")
+  @SlowReportQuery
+  fun getPrisoners(
+    @RequestParam(value = "includeAliases", required = false, defaultValue = "false")
+    @Parameter(description = "If true the result set should include a row for every matched alias.  If the request includes some combination of firstName, lastName and dateOfBirth then this will be a subset of the OFFENDERS records for one or more offenders. Otherwise it will be every OFFENDERS record for each match on the other search criteria. Default is false.")
+    includeAliases: Boolean,
```

Comment:
if the param has a `defaultValue` then I've mapped as a not nullable kotlin variable